### PR TITLE
Avoid the occurrence of 0 physical cpu cores

### LINKED ISSUE #6504: Avoid the occurrence of 0 physical cpu cores

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #6504

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hehechen — APPROVED** [review]

File: dbms/src/Common/getNumberOfCPUCores.cpp [hunk 1]
```diff
@@ -58,5 +58,6 @@ void computeAndSetNumberOfPhysicalCPUCores(UInt16 number_of_logical_cpu_cores_,
     // - `(hardware_logical_cpu_cores / number_of_hardware_physical_cores)` means how many logical cpu core a physical cpu core has.
     // - `number_of_logical_cpu_cores_ / (hardware_logical_cpu_cores / number_of_hardware_physical_cores)` means how many physical cpu cores the tiflash process could use. (Actually, it's needless to get physical cpu cores in virtual environment, but we must ensure the behavior `1` is not broken)
     auto hardware_logical_cpu_cores = std::thread::hardware_concurrency();
-    CPUCores::number_of_physical_cpu_cores = number_of_logical_cpu_cores_ / (hardware_logical_cpu_cores / number_of_hardware_physical_cores);
+    UInt16 physical_cpu_cores = number_of_logical_cpu_cores_ / (hardware_logical_cpu_cores / number_of_hardware_physical_cores);
+    CPUCores::number_of_physical_cpu_cores = physical_cpu_cores > 0 ? physical_cpu_cores : 1;
```
[the remark this answers is not in the document]
Comment (xzhangxian1008) [reply dbms/src/Common/getNumberOfCPUCores.cpp:62]: > Maybe we can add a log here.

okk

File: dbms/src/Common/getNumberOfCPUCores.cpp [hunk 2]
```diff
@@ -58,5 +58,15 @@ void computeAndSetNumberOfPhysicalCPUCores(UInt16 number_of_logical_cpu_cores_,
...
+    auto * log = &Poco::Logger::get("CPUCores");
```
Comment (SeaRise) [inline dbms/src/Common/getNumberOfCPUCores.cpp:63]:
```suggestion
    auto log = Logger::get("CPUCores");
```
Please use `<Common/Logger.h>`

Comment (xzhangxian1008) [reply dbms/src/Common/getNumberOfCPUCores.cpp]: > Please use `<Common/Logger.h>`

okk

**Review by SeaRise — APPROVED** [review]
lgtm

**Comment by xzhangxian1008** [comment]
> Maybe add some ut

I think ut is unnecessary here

**Pull request merged**